src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
+ * given the physical possibility. In m/s.
  *
+ * @param max_allowable_speed_profile [in] The array that will be modified in-place to
+ * contain the absolute maximum allowable speed profile. In m/s.
  */
-void app_trajectory_planner_getMaxAllowableSpeedProfile(
-    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    Polynomial2dOrder3_t path, const unsigned int num_elements,
-    ArcLengthParametrization_t arc_length_parameterization,
-    const float arc_segment_length, const float max_allowable_acceleration);
+void app_trajectory_planner_getAbsoluteMaximumSpeedProfile(
+    Polynomial2dOrder3_t path, unsigned int num_elements, float t_start, float t_end,
+    float max_allowable_acceleration, float speed_cap,
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS]);
 
 /**
- * Builds the forwards continuous velocity profile for the given parameters.
- * NOTE: The velocity profile returned by this function is NOT reverse continuous and does
- * not guarantee a feasible path to follow.
- *
- * NOTE: The velocity_profile will contain the same number of elements as the input
- * max_allowable_speed[] array
- *
- * @pre The velocity_profile array is pre-allocated to contain up to at least
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS elements
- *
- * @param num_segments [in] The number of segments(elements) in the
- * max_allowable_speed_profile input array
- * @param velocity_profile [out] The array that the forwards continuous array will be
- * copied into
- * @param max_allowable_speed_profile [in] The pre-allocated array that will contain all
- * of the maximum allowable velocity values for each point on the curve
- * @param arc_segment_length [in] The length of each arc length segment in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point in
- * m/s^2
- * @param max_allowable_speed [in] The max allowable speed at any point in m/s
- *
- * */
-void app_trajectory_planner_generateForwardsContinuousVelocityProfile(
-    const unsigned int num_segments,
-    float velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const float arc_segment_length, const float max_allowable_acceleration,
-    const float max_allowable_speed);
+ * Function generates the time profile for a speed profile with given segment lengths and
+ * start and end velocities. This is done using a constant acceleration assumption over
+ * the total displacement of each segment
+ *
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param segment_lengths [in] The length of each segment in meters. The segment length is
+ * the distance travelled between successive speed elements.
+ *
+ * @param speeds [in] The speed profile defining the current speed at each element. In
+ * m/s.
```

Comment:
Similar comment about this function being shared, and units relevant?

Reply:
Changed the unit comments to be more generic